packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -265,6 +275,29 @@ class TestCommand extends FlutterCommand {
       ];
     }
 
+    final int shardIndex = int.tryParse(stringArg('shard-index') ?? '');
+    if (shardIndex != null && (shardIndex < 0 || !shardIndex.isFinite)) {
+      throwToolExit(
+          'Could not parse --shard-index=$shardIndex argument. It must be an integer greater than -1.');
+    }
+
+    final int totalShards = int.tryParse(stringArg('total-shards') ?? '');
+    if (totalShards != null && (totalShards <= 0 || !totalShards.isFinite)) {
+      throwToolExit(
+          'Could not parse --total-shards=$totalShards argument. It must be an integer greater than zero.');
+    }
+
+    if (totalShards != null) {
+      if (shardIndex == null) {
+        throwToolExit(
+            'If you set --total-hards you need to add --shard-index.');
+      }
+    } else if (shardIndex != null) {
+      if (totalShards == null) {
+        throwToolExit('If you set --shard-index you need to add --total-hard.');
```

Comment:
```suggestion
        throwToolExit('If you set --shard-index you need to add --total-shards.');
```

Reply:
Thanks!

Reply:
I commited your suggestion and it looks like it uses your user for the commit, do you have a CLI? :P